core/trino-main/src/test/java/io/trino/sql/rewrite/TestStatementRewrite.java

```diff
@@ -0,0 +1,210 @@
+import io.trino.spi.connector.ConnectorMetadata;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.session.PropertyMetadata;
+import io.trino.spi.transaction.IsolationLevel;
+import io.trino.sql.SqlFormatterUtil;
+import io.trino.sql.analyzer.FeaturesConfig;
+import io.trino.sql.parser.ParsingOptions;
+import io.trino.sql.parser.SqlParser;
+import io.trino.sql.tree.Statement;
+import io.trino.testing.TestingMetadata;
+import io.trino.transaction.TransactionManager;
+import io.trino.sql.rewrite.StatementRewrite.Rewrite;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static io.trino.connector.CatalogName.createInformationSchemaCatalogName;
+import static io.trino.connector.CatalogName.createSystemTablesCatalogName;
+import static io.trino.cost.StatsCalculator.noopStatsCalculator;
+import static io.trino.execution.warnings.WarningCollector.NOOP;
+import static io.trino.metadata.MetadataManager.createTestMetadataManager;
+import static io.trino.operator.scalar.ApplyFunction.APPLY_FUNCTION;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.testing.TestingEventListenerManager.emptyEventListenerManager;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static io.trino.transaction.TransactionBuilder.transaction;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+
+public class TestStatementRewrite
+{
+    private static final String TPCH_CATALOG = "tpch";
+    private static final CatalogName TPCH_CATALOG_NAME = new CatalogName(TPCH_CATALOG);
+    private static final Session.SessionBuilder CLIENT_SESSION_BUILDER = testSessionBuilder()
+            .setCatalog(TPCH_CATALOG)
+            .setSchema("s1")
+            .addPreparedStatement("q1", "SELECT a, ? as col2 FROM tpch.s1.t1");
+    private static final Session SETUP_SESSION = testSessionBuilder()
+            .setCatalog("c1")
+            .setSchema("s1")
+            .build();
+
+    private static final SqlParser SQL_PARSER = new SqlParser();
+    private static final DescribeOutputRewrite DESCRIBE_OUTPUT_REWRITE = new DescribeOutputRewrite();
+
+    private TransactionManager transactionManager;
+    private AccessControlManager accessControl;
+    private Metadata metadata;
+
+    @Test
+    public void testDescribeOutputFormatSql()
+    {
+        transaction(transactionManager, accessControl)
+                .readUncommitted()
+                .execute(beginTransaction(CLIENT_SESSION_BUILDER), session -> {
+                    Statement root = SQL_PARSER.createStatement("DESCRIBE OUTPUT q1", new ParsingOptions());
```

Comment:
To make the test more to the point, you could create a `Statement` here instead of invoking the parser.

Reply:
I don't understand what you said about creating a `Statement`. I used the parser to create a `Statement` from a SQL string actually.
Could you explain more? Thanks.

Reply:
I meant that you can use a public constructor of `DescribeInput` / `DescribeOutput` to get an instance of those statements instead of parsing them from SQL. Parsing the `DESCRIBE ...` statements is not subject of this change, so no need to make it part of the test.

Reply:
ok, I got it. I'll fix them. Thank you.